File: pinyougou-user-web/src/main/webapp/js/service/userService.js
// 用户服务层
pyg.service('userService', function ($http) {

    // 查询所有数据
    this.findAll = function () {
        return $http.get('user/findAll.do');
    };

    // 无条件分页查询
    this.findPage = function (page, size) {
        return $http.get('user/findPage.do?page=' + page + '&size=' + size);
    };

    // 根据 ID 查询实体
    this.findOne = function (id) {
        return $http.get('user/findOne.do?id=' + id);
    };

    // 用户注册
    this.add = function (entity, smsCode) {
        return $http.post('user/add.do?smsCode=' + smsCode, entity);
    };

    // 修改
    this.update = function (entity) {
        return $http.post('user/update.do', entity);
    };

    // 批量删除
    this.batchDelete = function (selectedList) {
        return $http.get('user/batchDelete.do?ids=' + selectedList);
    };

    // 多条件分页查询
    this.findPageLimit = function (page, size, searchEntity) {
        return $http.post('user/search.do?page=' + page + "&size=" + size, searchEntity);
    };

    // 发送短信验证码
    this.sendSmsCode = function (phone) {
        return $http.get("user/sendSmsCode.do?phone=" + phone);
    };
});

File: pinyougou-user-interface/src/main/java/com/lsf/pinyougou/user/service/UserService.java
package com.lsf.pinyougou.user.service;

import java.util.List;

import com.lsf.pinyougou.pojo.TbAddress;
import com.lsf.pinyougou.pojo.TbUser;
import vo.PageResult;


/**
 * 用户服务层接口
 */
public interface UserService {

	/**
	 * 返回全部列表
	 */
	List<TbUser> findAll();
	
	
	/**
	 * 无条件分查询
	 */
	PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 多条件分页查询
	 */
	PageResult findPageLimit(TbUser user, int pageNum, int pageSize);
	
	
	/**
	 * 用户注册
	*/
	void add(TbUser user);
	
	
	/**
	 * 修改
	 */
	void update(TbUser user);
	

	/**
	 * 根据 ID 获取实体
	 */
	TbUser findOne(long id);
	
	
	/**
	 * 批量删除
	 */
	void batchDelete(Long[] ids);


	/**
	 * 为手机号 phone 生成短信验证码
	 */
	void createSmsCode(String phone);


	/**
	 * 判断手机号 phone 对应的验证码是否正确
	 */
	boolean checkSmsCode(String phone, String smsCode);



}
